fix(actions): secure-checkout fails on Windows — Git Bash mangles backslash paths

## Summary

The `secure-checkout` composite action fails on Windows runners because it references a bash script via a path containing backslashes. Git Bash interprets these as escape characters, producing a mangled path and exit code 127.

## Problem

On Windows runners, the `run:` directives in `action.yml` (lines 122 and 131) use:

```yaml
run: ${{ github.action_path }}/../../../scripts/ci/actions/secure-checkout.sh
```

`${{ github.action_path }}` is a GitHub Actions expression expanded by the **runner** before bash executes. On Windows, it produces literal backslashes:

```
D:\a\_actions\lgtm-hq\lgtm-ci\v0\.github\actions\secure-checkout
```

The full expanded `run:` value becomes:

```
D:\a\_actions\lgtm-hq\lgtm-ci\v0\.github\actions\secure-checkout/../../../scripts/ci/actions/secure-checkout.sh
```

This literal string is written to a temp `.sh` file and executed by Git Bash, which interprets backslashes as escape characters, producing:

```
D:a_actionslgtm-hqlgtm-civ0.github/actions/secure-checkout/../../../scripts/ci/actions/secure-checkout.sh
```

```
No such file or directory
##[error]Process completed with exit code 127.
```

**Impact:**
- All Windows builds that use `secure-checkout` fail at the checkout step.
- Downstream projects (e.g., Rustume) cannot produce Windows binaries in CI.
- Blocks any reusable workflow that includes Windows in its runner matrix (#69).

## Proposed Solution

Replace the `${{ github.action_path }}` expression (expanded pre-bash) with the `GITHUB_ACTION_PATH` environment variable (available at bash runtime), and normalize backslashes to forward slashes using bash parameter expansion.

**Before** (both steps on lines 122 and 131):
```yaml
run: ${{ github.action_path }}/../../../scripts/ci/actions/secure-checkout.sh
```

**After:**
```yaml
run: |
  SCRIPT="${GITHUB_ACTION_PATH//\\//}/../../../scripts/ci/actions/secure-checkout.sh"
  "$SCRIPT"
```

**Why this works:**
- `GITHUB_ACTION_PATH` is automatically set by GitHub Actions for composite actions — same value as `github.action_path` but accessible to bash at runtime.
- `${VAR//\\//}` is standard bash parameter expansion: replace all `\` with `/`. This is a no-op on Linux/macOS (paths already use `/`) and fixes Windows paths.
- Git Bash supports this syntax.

**Additionally:** Add BATS integration tests for `secure-checkout.sh` following the existing pattern used by `sign-artifact` and `verify-signature` test suites.

## References

- Downstream report: lgtm-hq/Rustume#100
- Blocks: #69 (reusable Rust cross-compilation workflow includes a Windows target)
- Affected file: `.github/actions/secure-checkout/action.yml` (lines 122, 131)
- Script: `scripts/ci/actions/secure-checkout.sh`

--- a/.github/actions/secure-checkout/action.yml
+++ b/.github/actions/secure-checkout/action.yml
@@ -119,7 +119,9 @@ runs:
         STEP: repo-dir
         WORKSPACE: ${{ github.workspace }}
         CHECKOUT_PATH: ${{ inputs.path }}
-      run: ${{ github.action_path }}/../../../scripts/ci/actions/secure-checkout.sh
+      run: |
+        SCRIPT="${GITHUB_ACTION_PATH//\\//}/../../../scripts/ci/actions/secure-checkout.sh"
+        "$SCRIPT"
 
     - name: Verify checkout integrity
       shell: bash
@@ -128,7 +130,9 @@ runs:
         STEP: verify
         REPOSITORY: ${{ inputs.repository }}
         PERSIST_CREDENTIALS: ${{ inputs['persist-credentials'] }}
-      run: ${{ github.action_path }}/../../../scripts/ci/actions/secure-checkout.sh
+      run: |
+        SCRIPT="${GITHUB_ACTION_PATH//\\//}/../../../scripts/ci/actions/secure-checkout.sh"
+        "$SCRIPT"
 
 branding:
   icon: "download"

--- a/tests/bats/integration/test_secure_checkout.bats
+++ b/tests/bats/integration/test_secure_checkout.bats
@@ -1,0 +1,173 @@
+#!/usr/bin/env bats
+# SPDX-License-Identifier: MIT
+# Purpose: Integration tests for secure-checkout action script
+
+load "../../helpers/common"
+load "../../helpers/mocks"
+load "../../helpers/github_env"
+
+SCRIPT="${PROJECT_ROOT}/scripts/ci/actions/secure-checkout.sh"
+
+setup() {
+	setup_temp_dir
+	save_path
+	setup_github_env
+	export LIB_DIR
+	export BATS_TEST_TMPDIR
+	export PROJECT_ROOT
+	export SCRIPT
+}
+
+teardown() {
+	restore_path
+	teardown_github_env
+	teardown_temp_dir
+}
+
+# =============================================================================
+# STEP validation tests
+# =============================================================================
+
+@test "secure-checkout: fails when STEP is not set" {
+	run bash -c 'unset STEP; bash "$SCRIPT" 2>&1'
+	assert_failure
+	assert_output --partial "STEP is required"
+}
+
+@test "secure-checkout: fails on unknown STEP" {
+	run bash -c 'export STEP=invalid; bash "$SCRIPT" 2>&1'
+	assert_failure
+	assert_output --partial "Unknown step"
+}
+
+# =============================================================================
+# STEP=repo-dir tests
+# =============================================================================
+
+@test "secure-checkout: repo-dir fails when WORKSPACE is not set" {
+	run bash -c '
+		export STEP=repo-dir
+		unset WORKSPACE
+		bash "$SCRIPT" 2>&1
+	'
+	assert_failure
+	assert_output --partial "WORKSPACE is required"
+}
+
+@test "secure-checkout: repo-dir outputs workspace when CHECKOUT_PATH is empty" {
+	run bash -c '
+		export STEP=repo-dir
+		export WORKSPACE="/home/runner/work/repo/repo"
+		export CHECKOUT_PATH=""
+		bash "$SCRIPT" 2>&1
+	'
+	assert_success
+	assert_github_output "path" "/home/runner/work/repo/repo"
+}
+
+@test "secure-checkout: repo-dir outputs workspace when CHECKOUT_PATH is unset" {
+	run bash -c '
+		export STEP=repo-dir
+		export WORKSPACE="/home/runner/work/repo/repo"
+		unset CHECKOUT_PATH
+		bash "$SCRIPT" 2>&1
+	'
+	assert_success
+	assert_github_output "path" "/home/runner/work/repo/repo"
+}
+
+@test "secure-checkout: repo-dir appends CHECKOUT_PATH to workspace" {
+	run bash -c '
+		export STEP=repo-dir
+		export WORKSPACE="/home/runner/work/repo/repo"
+		export CHECKOUT_PATH="subdir"
+		bash "$SCRIPT" 2>&1
+	'
+	assert_success
+	assert_github_output "path" "/home/runner/work/repo/repo/subdir"
+}
+
+# =============================================================================
+# STEP=verify tests
+# =============================================================================
+
+@test "secure-checkout: verify fails when REPOSITORY is not set" {
+	run bash -c '
+		export STEP=verify
+		unset REPOSITORY
+		bash "$SCRIPT" 2>&1
+	'
+	assert_failure
+	assert_output --partial "REPOSITORY is required"
+}
+
+@test "secure-checkout: verify fails when not in a git repository" {
+	run bash -c '
+		cd "$BATS_TEST_TMPDIR"
+		export STEP=verify
+		export REPOSITORY="owner/repo"
+		bash "$SCRIPT" 2>&1
+	'
+	assert_failure
+	assert_output --partial "Checkout verification failed"
+}
+
+@test "secure-checkout: verify succeeds in a git repository" {
+	setup_mock_git_repo
+
+	run bash -c '
+		cd "$MOCK_GIT_REPO"
+		export STEP=verify
+		export REPOSITORY="owner/repo"
+		bash "$SCRIPT" 2>&1
+	'
+	assert_success
+	assert_output --partial "Repository: owner/repo"
+	assert_output --partial "Commit:"
+}
+
+@test "secure-checkout: verify warns when credential helper is set with persist-credentials=false" {
+	setup_mock_git_repo
+	(cd "$MOCK_GIT_REPO" && git config credential.helper "store")
+
+	run bash -c '
+		cd "$MOCK_GIT_REPO"
+		export STEP=verify
+		export REPOSITORY="owner/repo"
+		export PERSIST_CREDENTIALS=false
+		bash "$SCRIPT" 2>&1
+	'
+	assert_success
+	assert_output --partial "Credential helper is configured despite persist-credentials=false"
+}
+
+@test "secure-checkout: verify does not warn when persist-credentials=true" {
+	setup_mock_git_repo
+	(cd "$MOCK_GIT_REPO" && git config credential.helper "store")
+
+	run bash -c '
+		cd "$MOCK_GIT_REPO"
+		export STEP=verify
+		export REPOSITORY="owner/repo"
+		export PERSIST_CREDENTIALS=true
+		bash "$SCRIPT" 2>&1
+	'
+	assert_success
+	refute_output --partial "Credential helper is configured"
+}
+
+@test "secure-checkout: verify does not warn when no credential helper is set" {
+	setup_mock_git_repo
+
+	run bash -c '
+		cd "$MOCK_GIT_REPO"
+		export GIT_CONFIG_GLOBAL=/dev/null
+		export GIT_CONFIG_NOSYSTEM=1
+		export STEP=verify
+		export REPOSITORY="owner/repo"
+		export PERSIST_CREDENTIALS=false
+		bash "$SCRIPT" 2>&1
+	'
+	assert_success
+	refute_output --partial "Credential helper is configured"
+}

